lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/components/RuntimeVersionManager.scala

```diff
@@ -206,8 +208,20 @@ class RuntimeVersionManager(
     */
   private def getEngine(version: SemVer): Try[Engine] = {
     val name = engineNameForVersion(version)
-    firstExisting(distributionManager.paths.engineSearchPaths.map(_ / name))
-      .map(loadEngine)
+    this.environment
+      .getEnvPath("ENSO_ENGINE_PATH")
+      .map { p =>
+        logger.info("Using explicit ENSO_ENGINE_PATH: " + p)
+        val manifest = loadAndCheckEngineManifest(p)
+        val engine   = Engine(version, p, manifest.get)
+        Success(engine)
```

Comment:
Does that mean that setting `ENSO_ENGINE_PATH` overrides the `version` that is _requested_?

Ok for development, but IMO we should check if the `version` provided by the found engine matches the requested one and if not, at least issue a warning.

Moreover, once we override the version, I don't think we should _pretend_ that it is a different version (which is currently done as the returned `Engine` reports the _requested_ version even if it _differs_ from the _actual_ version).

Reply:
Currently the version is extracted from the engine path, as the directory structure of Enso distribution assumes the directory for each engine is its version, i.e. `0.0.0-dev`. This will not necessarily match the structure in `built-distribution` or some custom settings though.

Maybe we should start including the engine version in `manifest.yaml`? Then we could more robustly verify/report the versions.

Reply:
> Ok for development

This is all meant only for development. That's also a reason why it is all documented in `CONTRIBUTING.md` and not in user facing documentation.

> robustly verify/report the versions.

This PR is supposed to _loosen robustness_, not tight it! The current development setup is complicated and my aim is to simplify it by `sbt runProjectManagerDistribution`. The goal is not to influence what happens in the production.